src/wallet/wallet.h

```diff
@@ -603,6 +603,18 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
     bool fFileBacked;
 
     std::set<int64_t> setKeyPool;
+
+    int64_t nTimeFirstKey;
+
+    /**
+     * Private version of AddWatchOnly method which does not accept a timestamp.
+     * Because this is an inherited virtual method, it is accessible despite
+     * being marked private, but it is marked private anyway to encourage use of
+     * the other AddWatchOnly which accepts a timestamp and sets nTimeFirstKey
+     * more intelligently for more efficient rescans.
```

Comment:
Ugh, I totally missed that its an override, thanks for catching that. Can you further document here that this method implies setting the first key time to 1?